python/tank_vendor/shotgun_authentication/user.py

```diff
@@ -0,0 +1,392 @@
+        cls._not_implemented("from_dict")
+
+    @classmethod
+    def _not_implemented(cls, method):
+        """
+        Raise a properly formatted error message when a method is not implemented.
+
+        :param method: Name of the method not implemented.
+
+        :raises NotImplementedError: Thrown with the message "<class-name>.<method-name>
+                                     is not implemented."
+        """
+        raise NotImplementedError(
+            "%s.%s is not implemented." % (cls.__name__, method)
+        )
+
+
+class SessionUser(ShotgunUser):
+    """
+    A user that authenticates to the Shotgun server using a session token.
+    """
+    def __init__(self, host, login, session_token, http_proxy, is_volatile=False):
+        """
+        Constructor.
+
+        :param host: Host for this Shotgun user.
+        :param login: Login name for the user.
+        :param session_token: Session token for the user.
+        :param http_proxy: HTTP proxy to use with this host. Defaults to None.
+        :param is_volatile: Indicates if the user can cache it's credentials to
+                            disk.
+        """
+
+        super(SessionUser, self).__init__(host, http_proxy)
+
+        self._login = login
+        self._session_token = session_token
+        self._is_volatile = is_volatile
+
+    def get_login(self):
+        """
+        Returns the login name for this user.
+
+        :returns: The login name string.
+        """
+        return self._login
+
+    def get_session_token(self):
+        """
+        Returns the session token for this user.
+
+        :returns: The session token string.
+        """
+        return self._session_token
+
+    def set_session_token(self, session_token):
+        """
+        Updates the session token for this user.
+
+        :param session_token: The new session token for this user.
+        """
+        self._session_token = session_token
+
+    def create_sg_connection(self):
+        """
+        Creates a Shotgun instance using the script user's credentials.
+
+        :returns: A Shotgun instance.
+        """
+        return _shotgun_instance_factory(
+            self._host, session_token=self._session_token, http_proxy=self._http_proxy,
+            user=self
+        )
+
+    def mark_volatile(self):
+        """
+        Marks this user as volatile. A volatile user won't save it's credentials to disk.
+        """
+        self._is_volatile = True
+
+    def is_volatile(self):
+        """
+        Returns if a user is volatile.
+
+        :returns: True if volatile, False otherwise.
+        """
+        return self._is_volatile
+
+    def save(self):
+        """
+        Saves a user's information in the local site cache.
+
+        :param user: Specifying a user to be the current user.
+
+        :raises CachingVolatileUserException: Raised if the user is volatile.
+        """
+        if self._is_volatile:
+            raise CachingVolatileUserException()
```

Comment:
why raise? why not just skip writing it? I have a feeling I might not totally understand this volatile concept :)